hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -61,10 +64,18 @@ public BaseActionExecutor(HoodieEngineContext context, HoodieWriteConfig config,
    *
    * @param metadata commit metadata of interest.
    */
-  protected final void writeTableMetadata(HoodieCommitMetadata metadata, HoodieData<WriteStatus> writeStatus, String actionType) {
+  protected final void writeTableMetadata(HoodieCommitMetadata metadata,
+                                          HoodieData<WriteStatus> writeStatus,
+                                          WriteOperationType operationType) {
     Option<HoodieTableMetadataWriter> metadataWriterOpt = table.getMetadataWriter(instantTime);
     if (metadataWriterOpt.isPresent()) {
       try (HoodieTableMetadataWriter metadataWriter = metadataWriterOpt.get()) {
+        // Recreate MDT for insert_overwrite_table operation.
+        if (WriteOperationType.INSERT_OVERWRITE_TABLE == operationType) {
+          HoodieTableMetadataUtil.deleteMetadataTable(table.getMetaClient(), table.getContext(), false);
+          new RunIndexActionExecutor<>(context, config, table, instantTime).execute();
```

Comment:
Since we are deleting the whole metadata table, I think it makes more sense to follow the usual `BaseHoodieWriteClient.initMetadataTable` (check its spark implementation). Let;s see if we can reuse that instead of using RunIndexActionExecutor.

Reply:
I checked the logic, it sounds the metadatawriter initialization will try to recreate the table. I will remove this `RunIndexActionExecutor` usage here to see if it works.